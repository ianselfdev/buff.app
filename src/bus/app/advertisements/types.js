export const types = {
    //Sync
    CREATE_AD_INSTANCE: 'CREATE_AD_INSTANCE',

    //Async
    CREATE_AD_INSTANCE_ASYNC: 'CREATE_AD_INSTANCE_ASYNC',
};
