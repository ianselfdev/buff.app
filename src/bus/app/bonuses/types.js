export const types = {
    //Sync
    FILL_AVAILABLE_BONUSES: 'FILL_AVAILABLE_BONUSES',

    //Async
    FETCH_AVAILABLE_BONUSES_ASYNC: 'FETCH_AVAILABLE_BONUSES_ASYNC',
    ACTIVATE_ALL_BONUSES_ASYNC: 'ACTIVATE_ALL_BONUSES_ASYNC',
    ACTIVATE_BONUS_ASYNC: 'ACTIVATE_BONUS_ASYNC',
};
