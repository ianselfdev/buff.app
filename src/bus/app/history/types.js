export const types = {
    //Sync
    FILL_HISTORY: 'FILL_HISTORY',
    ADD_HISTORY_FILTER_PARAMETER: 'ADD_HISTORY_FILTER_PARAMETER',
    REMOVE_HISTORY_FILTER_PARAMETER: 'REMOVE_HISTORY_FILTER_PARAMETER',
    CLEAR_HISTORY: 'CLEAR_HISTORY',

    //Async
    FETCH_HISTORY_ASYNC: 'FETCH_HISTORY_ASYNC',
    REMOVE_HISTORY_FILTER_PARAMETER_ASYNC: 'REMOVE_HISTORY_FILTER_PARAMETER_ASYNC',
    FILTER_HISTORY_ASYNC: 'FILTER_HISTORY_ASYNC',
};
