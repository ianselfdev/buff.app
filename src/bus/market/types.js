export const types = {
    //Sync
    FILL_MARKET_ITEMS: 'FILL_MARKET_ITEMS',
    FILL_USER_ITEMS: 'FILL_USER_ITEMS',
    FILL_GIFT_CODE: 'FILL_GIFT_CODE',
    ADD_FILTER_PARAMETER: 'ADD_FILTER_PARAMETER',
    REMOVE_FILTER_PARAMETER: 'REMOVE_FILTER_PARAMETER',

    //Async
    FETCH_MARKET_ITEMS_ASYNC: 'FETCH_MARKET_ITEMS_ASYNC',
    FETCH_USER_ITEMS_ASYNC: 'FETCH_USER_ITEMS_ASYNC',
    BUY_ITEM_ASYNC: 'BUY_ITEM_ASYNC',
    ACTIVATE_ITEM_ASYNC: 'ACTIVATE_ITEM_ASYNC',
    FILTER_MARKET_ITEMS_ASYNC: 'FILTER_MARKET_ITEMS_ASYNC',
    FILTER_USER_ITEMS_ASYNC: 'FILTER_USER_ITEMS_ASYNC',
    REMOVE_FILTER_PARAMETER_ASYNC: 'REMOVE_FILTER_PARAMETER_ASYNC',
};
