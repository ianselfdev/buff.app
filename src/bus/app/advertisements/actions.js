//Core
import { types } from './types';

export const advertisementActions = {
    createAdInstance: (node) => {
        return {
            type: types.CREATE_AD_INSTANCE,
            payload: node,
        };
    },

    createAdInstanceAsync: (node) => {
        return {
            type: types.CREATE_AD_INSTANCE_ASYNC,
            payload: node,
        };
    },
};
