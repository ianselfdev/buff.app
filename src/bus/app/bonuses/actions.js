//Types
import { types } from './types';

export const bonusesActions = {
    //Sync
    fillAvailableBonuses: (bonuses) => {
        return {
            type: types.FILL_AVAILABLE_BONUSES,
            payload: bonuses,
        };
    },

    //Async
    fetchAvailableBonusesAsync: () => {
        return {
            type: types.FETCH_AVAILABLE_BONUSES_ASYNC,
        };
    },
    activateAllBonusesAsync: () => {
        return {
            type: types.ACTIVATE_ALL_BONUSES_ASYNC,
        };
    },
    activateBonusAsync: (bonus) => {
        return {
            type: types.ACTIVATE_BONUS_ASYNC,
            payload: bonus,
        };
    },
};
