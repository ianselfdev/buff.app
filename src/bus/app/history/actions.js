//Types
import { types } from './types';

export const historyActions = {
    //Sync
    fillHistory: (history) => {
        return {
            type: types.FILL_HISTORY,
            payload: history,
        };
    },

    clearHistory: () => {
        return {
            type: types.CLEAR_HISTORY,
        };
    },

    addHistoryFilterParameter: (parametersObj) => {
        return {
            type: types.ADD_HISTORY_FILTER_PARAMETER,
            payload: parametersObj,
        };
    },

    removeHistoryFilterParameter: (parameter) => {
        return {
            type: types.REMOVE_HISTORY_FILTER_PARAMETER,
            payload: parameter,
        };
    },

    //Async
    fetchHistoryAsync: () => {
        return {
            type: types.FETCH_HISTORY_ASYNC,
        };
    },

    filterHistoryAsync: (parameter, value) => {
        return {
            type: types.FILTER_HISTORY_ASYNC,
            payload: {
                parameter,
                value,
            },
        };
    },

    removeHistoryFilterParameterAsync: (parameter) => {
        return {
            type: types.REMOVE_HISTORY_FILTER_PARAMETER_ASYNC,
            payload: parameter,
        };
    },
};
