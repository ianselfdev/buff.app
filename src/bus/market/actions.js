import { types } from './types';

export const marketActions = {
    //Sync
    fillMarketItems: (items) => {
        return {
            type: types.FILL_MARKET_ITEMS,
            payload: items,
        };
    },

    fillUserItems: (items) => {
        return {
            type: types.FILL_USER_ITEMS,
            payload: items,
        };
    },

    //Async
    fetchMarketItemsAsync: () => {
        return {
            type: types.FETCH_MARKET_ITEMS_ASYNC,
        };
    },

    fetchUserItemsAsync: () => {
        return {
            type: types.FETCH_USER_ITEMS_ASYNC,
        };
    },

    buyItemAsync: (itemId) => {
        return {
            type: types.BUY_ITEM_ASYNC,
            payload: itemId,
        };
    },
};
