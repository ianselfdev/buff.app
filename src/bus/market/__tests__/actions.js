//Actions
import { marketActions } from '../actions';

//Types
import { types } from '../types';

//Instruments
const market = {
    items: [{}, {}, {}, {}],
    itemId: '1',
    giftCode: '1234-1234-1234-1234',
    parametersObj: { parameter: 'parameter', value: 'value' },
    parameter: 'parameter',
};

describe('market actions: ', () => {
    test('fillMarketItems', () => {
        expect(marketActions.fillMarketItems(market.items)).toMatchInlineSnapshot(`
Object {
  "payload": Array [
    Object {},
    Object {},
    Object {},
    Object {},
  ],
  "type": "FILL_MARKET_ITEMS",
}
`);
    });
    test('fillUserItems', () => {
        expect(marketActions.fillUserItems(market.items)).toMatchInlineSnapshot(`
Object {
  "payload": Array [
    Object {},
    Object {},
    Object {},
    Object {},
  ],
  "type": "FILL_USER_ITEMS",
}
`);
    });
    test('fillGiftCode', () => {
        expect(marketActions.fillGiftCode(market.giftCode)).toMatchInlineSnapshot(`
Object {
  "payload": "1234-1234-1234-1234",
  "type": "FILL_GIFT_CODE",
}
`);
    });
    test('addMarketFilterParameter', () => {
        expect(marketActions.addMarketFilterParameter(market.parametersObj)).toMatchInlineSnapshot(`
Object {
  "payload": Object {
    "parameter": "parameter",
    "value": "value",
  },
  "type": "ADD_MARKET_FILTER_PARAMETER",
}
`);
    });
    test('removeMarketFilterParameter', () => {
        expect(marketActions.removeMarketFilterParameter(market.parameter)).toMatchInlineSnapshot(`
Object {
  "payload": "parameter",
  "type": "REMOVE_MARKET_FILTER_PARAMETER",
}
`);
    });

    test('fetchMarketItemsAsync', () => {
        expect(marketActions.fetchMarketItemsAsync()).toMatchInlineSnapshot(`
Object {
  "type": "FETCH_MARKET_ITEMS_ASYNC",
}
`);
    });
    test('fetchUserItemsAsync', () => {
        expect(marketActions.fetchUserItemsAsync()).toMatchInlineSnapshot(`
Object {
  "type": "FETCH_USER_ITEMS_ASYNC",
}
`);
    });
    test('buyItemAsync', () => {
        expect(marketActions.buyItemAsync(market.itemId)).toMatchInlineSnapshot(`
Object {
  "payload": "1",
  "type": "BUY_ITEM_ASYNC",
}
`);
    });
    test('activateItemAsync', () => {
        expect(marketActions.activateItemAsync(market.itemId)).toMatchInlineSnapshot(`
Object {
  "payload": "1",
  "type": "ACTIVATE_ITEM_ASYNC",
}
`);
    });
    test('removeMarketFilterParameterAsync', () => {
        expect(marketActions.removeMarketFilterParameterAsync(market.parameter))
            .toMatchInlineSnapshot(`
Object {
  "payload": "parameter",
  "type": "REMOVE_MARKET_FILTER_PARAMETER_ASYNC",
}
`);
    });
    test('filterMarketItemsAsync', () => {
        expect(
            marketActions.filterMarketItemsAsync(
                market.parametersObj.parameter,
                market.parametersObj.value,
            ),
        ).toEqual({
            type: types.FILTER_MARKET_ITEMS_ASYNC,
            payload: market.parametersObj,
        });
    });
    test('filterUserItemsAsync', () => {
        expect(
            marketActions.filterUserItemsAsync(
                market.parametersObj.parameter,
                market.parametersObj.value,
            ),
        ).toEqual({
            type: types.FILTER_USER_ITEMS_ASYNC,
            payload: market.parametersObj,
        });
    });
});
