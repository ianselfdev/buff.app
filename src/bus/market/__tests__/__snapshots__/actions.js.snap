// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`market actions:  activateItemAsync 1`] = `
Object {
  "payload": "1",
  "type": "ACTIVATE_ITEM_ASYNC",
}
`;

exports[`market actions:  addFilterParameter 1`] = `
Object {
  "payload": Object {
    "parameter": "parameter",
    "value": "value",
  },
  "type": "ADD_FILTER_PARAMETER",
}
`;

exports[`market actions:  buyItemAsync 1`] = `
Object {
  "payload": "1",
  "type": "BUY_ITEM_ASYNC",
}
`;

exports[`market actions:  fetchMarketItemsAsync 1`] = `
Object {
  "type": "FETCH_MARKET_ITEMS_ASYNC",
}
`;

exports[`market actions:  fetchUserItemsAsync 1`] = `
Object {
  "type": "FETCH_USER_ITEMS_ASYNC",
}
`;

exports[`market actions:  fillGiftCode 1`] = `
Object {
  "payload": "1234-1234-1234-1234",
  "type": "FILL_GIFT_CODE",
}
`;

exports[`market actions:  fillMarketItems 1`] = `
Object {
  "payload": Array [
    Object {},
    Object {},
    Object {},
    Object {},
  ],
  "type": "FILL_MARKET_ITEMS",
}
`;

exports[`market actions:  fillUserItems 1`] = `
Object {
  "payload": Array [
    Object {},
    Object {},
    Object {},
    Object {},
  ],
  "type": "FILL_USER_ITEMS",
}
`;

exports[`market actions:  removeFilterParameter 1`] = `
Object {
  "payload": "parameter",
  "type": "REMOVE_FILTER_PARAMETER",
}
`;

exports[`market actions:  removeFilterParameterAsync 1`] = `
Object {
  "payload": "parameter",
  "type": "REMOVE_FILTER_PARAMETER_ASYNC",
}
`;
